### PR TITLE
add(track demux logs in housekeeper)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Demux logs are no longer tracked when cgstats is deprecated. This PR adds logic to track them in housekeeper as propred here: #2217

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/meta/demultiplex/test_demux_post_processing.py [hunk 1]
```diff
@@ -870,3 +865,40 @@ def test_copy_sample_sheet(demultiplex_context: CGConfig):
...
+    # WHEN adding the demux logs to housekeeper
+    demux_post_processing_api.add_demux_logs_to_housekeeper(flow_cell=dragen_flow_cell)
+
+    # THEN the demux log was added to housekeeper
+    files = demux_post_processing_api.hk_api.get_files(
+        tags=[SequencingFileTag.DEMUX_LOG],
+        bundle=dragen_flow_cell.id,
+    ).all()
+
+    assert len(files) == 2
```
Comment (diitaz93) [inline tests/meta/demultiplex/test_demux_post_processing.py:904]: I would do an additional checking with the names of the files

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

**Comment by ChrOertlin** [comment]
Tested:
image

File: cg/meta/demultiplex/demux_post_processing.py [hunk 2]
```diff
@@ -42,7 +41,7 @@
 from cg.store import Store
 from cg.store.models import Flowcell, SampleLaneSequencingMetrics
 from cg.utils import Process
-from cg.utils.files import get_file_in_directory
+from cg.utils.files import get_file_in_directory, get_files_matching_pattern
 from cg.constants.demultiplexing import DemultiplexingDirsAndFiles
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/meta/demultiplex/demux_post_processing.py:45]: moved it

**Pull request merged**